src/tools/rbd_nbd/rbd-nbd.cc

```diff
@@ -90,6 +92,20 @@ static bool exclusive = false;
 #endif
 #define htonll(a) ntohll(a)
 
+static int do_unmap();
+
+static void handle_signal(int signum)
+{
+  assert(signum == SIGINT || signum == SIGTERM);
+  derr << "*** Got signal " << sig_str(signum) << " ***" << dendl;
+  dout(20) << __func__ << ": " << "sending NBD_DISCONNECT" << dendl;
+  if (ioctl(nbd, NBD_DISCONNECT) < 0) {
+    derr << "rbd-nbd: disconnect failed" << cpp_error(errno) << std::endl;
+  } else {
+    dout(20) << __func__ << ": " << "disconnected" << dendl;
+  }
```

Comment:
@liupan1111  @trociny  Let me try to make this discussion easier, Please consider this scenario, a device is mapped, but we don't have permission to unmap it.

Then a user call a do_unmap(),  but we error out with the message of "the device is not used", it is not proper, do you agree?